### ISSUE TITLE
Possibility to provide object dependent map for CommitPropertiesProvider.

### ISSUE THREAD

**Issue by chioai1309** [javers/javers#870]
According to open question state here https://stackoverflow.com/questions/57518008/is-there-possibility-to-provide-object-dependent-map-for-commitpropertiesprovide.

We would like to have a domain object parameter for the CommitPropertiesProvider in order to provide the commit properties for context dependent in order to using [commit-property-filter](https://javers.org/documentation/jql-examples/#commit-property-filter)  feature later on. 

```
public class CustomCommitPropertiesProvider implements CommitPropertiesProvider {
   public Map<String, String> provide(Object domainObject) {
        if (domainObject instanceof ModerationEntity) {
            // return map with key = "listId" & value = ModerationEntity#listId
        }
        // return emptyMap
    }
}
```

**Comment by bartoszwalacik** [comment]
Makes sense, we are waiting for your PR :)

**Comment by chioai1309** [comment]
The PR is created. Sorry there's a mistake on the commit message refer to the wrong issue number (860 instead of 870)

**Comment by chioai1309** [comment]
#873 https://github.com/javers/javers/pull/873

**Comment by bartoszwalacik** [comment]
ok, i'm working on it. the biggest issue is keeping backward compatibility

**Comment by bartoszwalacik** [comment]
released in 5.7.0, @chioai1309 thanks for your contribution!

**Comment by chioai1309** [comment]
@bartoszwalacik : you're welcome. Thank you very much for the quick release. 👍
